packages/amplication-server/src/core/project/project.service.ts

```diff
@@ -263,6 +263,32 @@ export class ProjectService {
     };
   }
 
+  private async shouldBlockBuild(userId: string): Promise<boolean> {
+    if (!this.billingService.isBillingEnabled) {
+      return false;
+    }
+
+    const workspace = await this.prisma.user
+      .findUnique({
+        where: {
+          id: userId,
+        },
+      })
+      .workspace();
+
+    if (!workspace) {
+      throw new Error("workspace not found");
```

Comment:
do we want to throw error and stop the service or we just want to make it clear that there is no workspace and continue